# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientChannelHelper.cpp
# Commit: ce5c3283167c
# Full Hash: ce5c3283167cd3bf9e6bc939418ecd390fc3d09f
# Author: Liang-Heng Chen <xeonchen@gmail.com>
# Date: 2019-11-29 04:58:13
# Description:
#   Bug 1599083 - remove old ClientSource from ClientManagerService before adding new one; r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D55100
# ==============================================================================

diff -r eb5dc7cac72a -r ce5c3283167c dom/clients/manager/ClientChannelHelper.cpp
--- a/dom/clients/manager/ClientChannelHelper.cpp	Thu Nov 28 23:09:19 2019 +0000
+++ b/dom/clients/manager/ClientChannelHelper.cpp	Thu Nov 28 20:03:57 2019 +0000
@@ -90,6 +90,9 @@
         Maybe<ClientInfo> newClientInfo = newLoadInfo->GetReservedClientInfo();
         if (newClientInfo) {
           if (!reservedClient || reservedClient->Info() != *newClientInfo) {
+            // clear `reservedClient` first to ensure the same clientInfo ID has
+            // been removed before adding again.
+            reservedClient.reset(nullptr);
             reservedClient = ClientManager::CreateSourceFromInfo(*newClientInfo,
                                                                  mEventTarget);
           }
